src/NServiceBus.MongoDB/SagaPersister/MongoSagaPersister.cs

```diff
@@ -177,8 +177,8 @@ private static void CheckUniqueProperty(IContainSagaData sagaData, KeyValuePair<
             Contract.Requires(!string.IsNullOrWhiteSpace(property));
             Contract.Requires(value != null);
 
-            var query = Query.EQ(property, value.ToString());
-
+            var query = (value is Guid) ? Query.EQ(property, (Guid)value) : Query.EQ(property, value.ToString());
```

Comment:
Looks like this was [already fixed](https://github.com/sbmako/NServiceBus.MongoDB/blob/61a81d2c1f3d5e128d2a700239d7b551d0e77dfc/src/NServiceBus.MongoDB/SagaPersister/MongoSagaPersister.cs#L184) (though it requires upgrading to NServiceBus 5).